### PR TITLE
Clarify that v1beta1 CRDs have to opt-in in pruning feature

### PR DESCRIPTION
**Pull request by yue9944882**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.17 Features: set Milestone to 1.17 and Base Branch to dev-1.17
> 
> For pull requests on Chinese localization, set Base Branch to release-1.15
>
> For pull requests on Korean Localization: set Base Branch to dev-1.16-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

File: content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md [hunk 1]
```diff
@@ -874,7 +874,7 @@ OpenAPI v2 Publishing is available as beta since 1.15, and as alpha since 1.14.
 {{< /note >}}
 
 With the OpenAPI v2 Publishing feature enabled, CustomResourceDefinition [OpenAPI v3 validation schemas](#validation) which are [structural](#specifying-a-structural-schema) are published as part
```
Comment (liggitt) [inline content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md]:
```suggestion
With the OpenAPI v2 Publishing feature enabled, CustomResourceDefinition [OpenAPI v3 validation schemas](#validation) which are [structural](#specifying-a-structural-schema) and [enable pruning](#preserving-unknown-fields) (opt-in in v1beta1, enabled by default in v1) are published as part
```

Comment (sttts) [reply content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md]: Logically correct, but incomplete. Pruning is not required for publishing.

Comment (liggitt) [reply content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md]: it is required to get a complete schema published. remember https://github.com/kubernetes/kubernetes/pull/82666?

Comment (sttts) [reply content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md]: Had to verify that we really went so far to require pruning. We did 🎉

**Comment by zacharysarah** [comment]
@yue9944882 Thanks for the update. ✨

@liggitt 👋 Does this look good to you?

/assign

**Comment by liggitt** [comment]
I made a suggestion with a link I think might be clearer

File: content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md (hunk 1 above)
Comment (yue9944882) [reply content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md]: yay

**Pull request merged**